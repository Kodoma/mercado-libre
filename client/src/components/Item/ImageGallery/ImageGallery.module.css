.carousel {
   width: 350px;
   margin-bottom: 10%;
}
.custom-tag {
    width: 350px;
    height: 350px;
    background: black;
}

.control {
    color: #333333;
}

@media (max-width: 812px) {
    .carousel {
        max-width: 300px;
        max-width: 500px;
    }
}