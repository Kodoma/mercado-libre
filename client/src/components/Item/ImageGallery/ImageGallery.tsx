import React from 'react';
import Slider from 'react-slick';
import styles from './ImageGallery.module.css';

const ImageGallery: React.FunctionComponent<ItemsGallery> = ({
    pictures
}: ItemsGallery): JSX.Element => {
    const slides =
        pictures &&
        pictures.map((i) => {
            return <img src={i.src} />;
        });

    const settings = {
        dots: true,
        infinite: true,
        speed: 500,
        slidesToShow: 1,
        slidesToScroll: 1,
        adaptiveHeight: true,
        className: styles.carousel
    };

    return <Slider {...settings}>{slides}</Slider>;
};

export default ImageGallery;

interface ItemsGallery {
    pictures: Pictures[];
}

interface Pictures {
    src: string;
}
