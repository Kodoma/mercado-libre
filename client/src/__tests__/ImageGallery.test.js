import React from 'react';
import { shallow } from 'enzyme';
import ImageGallery from "../components/Item/ImageGallery/ImageGallery";

describe('__tests__ ImageGallery', () => {
    it('should render my ImageGallery component', () => {
        const component = shallow(<ImageGallery  pictures={[]}/>);
        expect(component.getElements()).toMatchSnapshot();
    });
});