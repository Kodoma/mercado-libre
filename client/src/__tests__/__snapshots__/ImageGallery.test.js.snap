// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__tests__ ImageGallery should render my ImageGallery component 1`] = `
Array [
  <Slider
    adaptiveHeight={true}
    className="carousel"
    dots={true}
    infinite={true}
    slidesToScroll={1}
    slidesToShow={1}
    speed={500}
  />,
]
`;
